app/components/clock_component.rb

```diff
@@ -0,0 +1,81 @@
+class ClockComponent < ViewComponent::Base
+  include Motion::Component
+  HOURS = 12
+  MINUTES = 60
+  SECONDS = 60
+  DEGREES = 360
+
+  stream_from "time:updated", :handle_time
```

Comment:
With the periodic timer API, this won't matter but just so that you know. If you are going to spawn a thread for every component, you would probably also want to use a different topic for each component. Right now, all components are going to process `time:updated` broadcasts for all of each others threads.

```ruby
  attr_reader :time_updated_topic

  def initialize(time: Time.now.utc)
    # ...

    @time_updated_topic = "time:updated:#{SecureRandom.uuid}"
    stream_from "time:updated:#{time_updated_topic}"
  end

  # ...

  def trigger_timer
    @to_time = @time + duration

    Thread.new(self) do |component|
      while Time.now < @to_time
        ActionCable.server.broadcast(component.time_updated_topic, { time: Time.now.utc })
        sleep 1
      end
    end
  end
```

Reply:
I addressed this to create a channel for every connection.